app/CocoaPods/CPHomeWindowController.m

```diff
@@ -137,4 +141,52 @@ - (CPCLIToolInstallationController *)createCLIToolInstallationController;
   return [CPCLIToolInstallationController controllerWithSuggestedDestinationURL:destinationURL];
 }
 
+
+// MARK: - Drag & Drop
+
+- (NSDragOperation)draggingEntered:(id<NSDraggingInfo>)sender
+{
+  NSPasteboard *pboard = [sender draggingPasteboard];
+  NSDragOperation sourceMask = [sender draggingSourceOperationMask];
+
+  if ([[pboard types] containsObject:NSFilenamesPboardType]) {
+    if (sourceMask & NSDragOperationLink) {
+      NSArray *files = [pboard propertyListForType:NSFilenamesPboardType];
+      // allow to drop files if any of them is Podfile
+      for (NSString *fileName in files) {
+        if ([[fileName lastPathComponent] isEqualToString:@"Podfile"]) {
+          return NSDragOperationCopy;
```

Comment:
Alright this looks good, any chance you can create a function that returns the first `Podfile` in a dragging pasteboard? then it could be used in both entered, and perform drag operation, this significantly reduces the amount of indentation for these methods and lets them talk about the what rather than the how.